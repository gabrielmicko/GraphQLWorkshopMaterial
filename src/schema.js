/**
 * Import makeExecutableSchema, resolvers, typeDefs
 * Call & export makeExecutableSchema, pass the imported variables
 */
import { makeExecutableSchema } from 'graphql-tools';
import resolvers from './resolvers';
import typeDefs from './types';

export default makeExecutableSchema({
  resolvers,
  typeDefs
});
