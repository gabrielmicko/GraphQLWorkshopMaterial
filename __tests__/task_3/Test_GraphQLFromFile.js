import { graphql } from 'graphql';
import schema from '../../src/schema';

test('GraphQL server should return all the speakers with talks.', () => {
  expect.assertions(1);
  const query = /* GraphQL */ `
    {
      speakers {
        talks
      }
    }
  `;
  return graphql(schema, query).then(result => {
    expect(result).toMatchSnapshot();
  });
});
