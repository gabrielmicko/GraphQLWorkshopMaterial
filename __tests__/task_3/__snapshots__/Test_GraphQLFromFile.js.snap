// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL server should return all the speakers with talks. 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Cannot query field "talks" on type "Speaker".],
  ],
}
`;
